crates/optimism/rpc/src/eth/pending_block.rs

```diff
@@ -45,6 +45,15 @@ where
         )>,
         Self::Error,
     > {
+        if let Some(rx) = self.inner.flashblocks_rx.as_ref() {
+            if let Some(block) = rx.borrow().as_ref() {
+                return Ok(Some((
+                    block.recovered_block.clone(),
+                    Arc::new(block.execution_output.receipts.iter().flatten().cloned().collect()),
```

Comment:
should we prep this in the type of the channel

`FlashBlockRx<N::Primitives>`

basically what we do for 

https://github.com/paradigmxyz/reth/blob/87647b25acf43689deb4568fdd98c6913a8d1819/crates/rpc/rpc-eth-types/src/pending_block.rs#L91-L97

then we can avoid doing this per request and only have to prep it once?